diff --git a/core/lib/Drupal/Core/DependencyInjection/DeprecatedServicePropertyTrait.php b/core/lib/Drupal/Core/DependencyInjection/DeprecatedServicePropertyTrait.php
index d77fa21..8b24e3a 100644
--- a/core/lib/Drupal/Core/DependencyInjection/DeprecatedServicePropertyTrait.php
+++ b/core/lib/Drupal/Core/DependencyInjection/DeprecatedServicePropertyTrait.php
@@ -24,6 +24,10 @@ public function __get($name) {
       @trigger_error("The property $name ($service_name service) is deprecated in $class_name and will be removed before Drupal 10.0.0.", E_USER_DEPRECATED);
       return \Drupal::service($service_name);
     }
+
+    if ($database) {
+      @trigger_error('$database parameter is deprecated in drupal:9.2.0 and is removed from drupal:10.0.0.');
+    }
   }
 
 }
diff --git a/core/modules/node/src/Plugin/views/argument/Vid.php b/core/modules/node/src/Plugin/views/argument/Vid.php
index ff1c0ac..073ac4f 100644
--- a/core/modules/node/src/Plugin/views/argument/Vid.php
+++ b/core/modules/node/src/Plugin/views/argument/Vid.php
@@ -2,7 +2,6 @@
 
 namespace Drupal\node\Plugin\views\argument;
 
-use Drupal\Core\Database\Connection;
 use Drupal\views\Plugin\views\argument\NumericArgument;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Drupal\node\NodeStorageInterface;
@@ -14,13 +13,6 @@
  */
 class Vid extends NumericArgument {
 
-  /**
-   * Database Service Object.
-   *
-   * @var \Drupal\Core\Database\Connection
-   */
-  protected $database;
-
   /**
    * The node storage.
    *
@@ -37,15 +29,11 @@ class Vid extends NumericArgument {
    *   The plugin_id for the plugin instance.
    * @param mixed $plugin_definition
    *   The plugin implementation definition.
-   * @param \Drupal\Core\Database\Connection $database
-   *   Database Service Object.
    * @param \Drupal\node\NodeStorageInterface $node_storage
    *   The node storage.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, Connection $database, NodeStorageInterface $node_storage) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, NodeStorageInterface $node_storage) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
-
-    $this->database = $database;
     $this->nodeStorage = $node_storage;
   }
 
@@ -57,7 +45,6 @@ public static function create(ContainerInterface $container, array $configuratio
       $configuration,
       $plugin_id,
       $plugin_definition,
-      $container->get('database'),
       $container->get('entity_type.manager')->getStorage('node')
     );
   }
@@ -68,17 +55,13 @@ public static function create(ContainerInterface $container, array $configuratio
   public function titleQuery() {
     $titles = [];
 
-    $results = $this->database->query('SELECT nr.vid, nr.nid, npr.title FROM {node_revision} nr WHERE nr.vid IN ( :vids[] )', [':vids[]' => $this->value])->fetchAllAssoc('vid', PDO::FETCH_ASSOC);
-    $nids = [];
-    foreach ($results as $result) {
-      $nids[] = $result['nid'];
-    }
-
-    $nodes = $this->nodeStorage->loadMultiple(array_unique($nids));
+    $results = $this->nodeStorage->getAggregateQuery()
+      ->allRevisions()
+      ->groupBy('title')
+      ->execute();
 
     foreach ($results as $result) {
-      $nodes[$result['nid']]->set('title', $result['title']);
-      $titles[] = $nodes[$result['nid']]->label();
+      $titles[] = $result['title'];
     }
 
     return $titles;
