diff --git a/core/modules/forum/src/Controller/ForumController.php b/core/modules/forum/src/Controller/ForumController.php
index 347013b..f44841e 100644
--- a/core/modules/forum/src/Controller/ForumController.php
+++ b/core/modules/forum/src/Controller/ForumController.php
@@ -294,8 +294,9 @@ protected function buildActionLinks($vid, TermInterface $forum_term = NULL) {
       if ($this->nodeAccess->createAccess($type)) {
         $node_type = $this->nodeTypeStorage->load($type);
         $links[$type] = [
-          '#attributes' => ['class' => ['action-links']],
           '#theme' => 'menu_local_action',
+          '#prefix' => '<ul class="action-links">',
+          '#suffix' => '</ul>',
           '#link' => [
             'title' => $this->t('Add new @node_type', [
               '@node_type' => $this->nodeTypeStorage->load($type)->label(),
@@ -323,8 +324,9 @@ protected function buildActionLinks($vid, TermInterface $forum_term = NULL) {
       // Anonymous user does not have access to create new topics.
       else {
         $links['login'] = [
-          '#attributes' => ['class' => ['action-links']],
           '#theme' => 'menu_local_action',
+          '#prefix' => '<ul class="action-links">',
+          '#suffix' => '</ul>',
           '#link' => [
             'title' => $this->t('Log in to post new content in the forum.'),
             'url' => Url::fromRoute('user.login', [], ['query' => $this->getDestinationArray()]),
