diff --git a/core/modules/block/src/Plugin/migrate/process/BlockPluginId.php b/core/modules/block/src/Plugin/migrate/process/BlockPluginId.php
index 448035d..6c337d3 100644
--- a/core/modules/block/src/Plugin/migrate/process/BlockPluginId.php
+++ b/core/modules/block/src/Plugin/migrate/process/BlockPluginId.php
@@ -76,7 +76,7 @@ public function transform($value, MigrateExecutableInterface $migrate_executable
       list($module, $delta) = $value;
       switch ($module) {
         case 'aggregator':
-          list($type) = explode('-', $delta);
+        list($type) = explode('-', $delta)[0];
           if ($type == 'feed') {
             return 'aggregator_feed_block';
           }
diff --git a/core/modules/block/tests/src/Functional/BlockTest.php b/core/modules/block/tests/src/Functional/BlockTest.php
index 4d804f1..3e2e89d 100644
--- a/core/modules/block/tests/src/Functional/BlockTest.php
+++ b/core/modules/block/tests/src/Functional/BlockTest.php
@@ -158,7 +158,7 @@ public function testAddBlockFromLibraryWithWeight() {
       $this->assertCount(1, $links, 'Found one matching link.');
       $this->assertEqual(t('Place block'), $links[0]->getText(), 'Found the expected link text.');
 
-      $query_string = explode('?', $links[0]->getAttribute('href'), 2);
+      $query_string = explode('?', $links[0]->getAttribute('href'), 2)[1];
       parse_str($query_string[1], $query_parts);
       $this->assertEqual($weight, $query_parts['weight'], 'Found the expected weight query string.');
 
