diff --git a/core/modules/system/tests/src/Functional/DrupalKernel/ContentNegotiationTest.php b/core/modules/system/tests/src/Functional/DrupalKernel/ContentNegotiationTest.php
index a57b941..52ec453 100644
--- a/core/modules/system/tests/src/Functional/DrupalKernel/ContentNegotiationTest.php
+++ b/core/modules/system/tests/src/Functional/DrupalKernel/ContentNegotiationTest.php
@@ -23,28 +23,32 @@ class ContentNegotiationTest extends BrowserTestBase {
    * expected.
    *
    * @see https://www.drupal.org/node/1716790
+   *
+   * @dataProvider providerTestBogusAcceptHeader
+   */
+  public function testBogusAcceptHeader($case, $header) {
+    $this->drupalGet('', [], ['Accept: ' . $header]);
+  }
+
+  /**
+   * Provides a list of headers to test.
    */
-  public function testBogusAcceptHeader() {
-    $tests = [
+  public function providerTestBogusAcceptHeader() {
+    return [
       // See https://bugs.webkit.org/show_bug.cgi?id=27267.
-      'Firefox 3.5 (2009)' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
-      'IE8 (2009)' => 'image/gif, image/jpeg, image/pjpeg, image/pjpeg, application/x-shockwave-flash, application/xaml+xml, application/vnd.ms-xpsdocument, application/x-ms-xbap, application/x-ms-application, application/vnd.ms-excel, application/vnd.ms-powerpoint, application/msword, application/x-silverlight, */*',
-      'Opera (2009)' => 'text/html, application/xml;q=0.9, application/xhtml+xml, image/png, image/jpeg, image/gif, image/x-xbitmap, */*;q=0.1',
-      'Chrome (2009)' => 'application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5',
+      ['Firefox 3.5 (2009)', 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8'],
+      ['IE8 (2009)', 'image/gif, image/jpeg, image/pjpeg, image/pjpeg, application/x-shockwave-flash, application/xaml+xml, application/vnd.ms-xpsdocument, application/x-ms-xbap, application/x-ms-application, application/vnd.ms-excel, application/vnd.ms-powerpoint, application/msword, application/x-silverlight, */*'],
+      ['Opera (2009)', 'text/html, application/xml;q=0.9, application/xhtml+xml, image/png, image/jpeg, image/gif, image/x-xbitmap, */*;q=0.1'],
+      ['Chrome (2009)', 'application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5'],
       // See https://github.com/symfony/symfony/pull/564.
-      'Firefox 3.6 (2010)' => 'text/html,application/xhtml+xml,application/json,application/xml;q=0.9,*/*;q=0.8',
-      'Safari (2010)' => '*/*',
-      'Opera (2010)' => 'image/jpeg,image/gif,image/x-xbitmap,text/html,image/webp,image/png,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.1',
+      ['Firefox 3.6 (2010)', 'text/html,application/xhtml+xml,application/json,application/xml;q=0.9,*/*;q=0.8'],
+      ['Safari (2010)', '*/*'],
+      ['Opera (2010)', 'image/jpeg,image/gif,image/x-xbitmap,text/html,image/webp,image/png,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.1'],
       // See https://www.drupal.org/node/1716790.
-      'Safari (2010), iOS 4.2.1 (2012)' => 'application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5',
-      'Android #1 (2012)' => 'application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5',
-      'Android #2 (2012)' => 'text/xml,text/html,application/xhtml+xml,image/png,text/plain,*/*;q=0.8',
+      ['Safari (2010), iOS 4.2.1 (2012)', 'application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5'],
+      ['Android #1 (2012)', 'application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5'],
+      ['Android #2 (2012)', 'text/xml,text/html,application/xhtml+xml,image/png,text/plain,*/*;q=0.8'],
     ];
-    foreach ($tests as $case => $header) {
-      $this->drupalGet('', [], ['Accept: ' . $header]);
-      $this->assertNoText('Unsupported Media Type', '"Unsupported Media Type" not found for ' . $case);
-      $this->assertText(t('Log in'), '"Log in" found for ' . $case);
-    }
   }
 
 }
