diff --git a/core/modules/tour/tour.module b/core/modules/tour/tour.module
index e6bbd2a..75e3d91 100644
--- a/core/modules/tour/tour.module
+++ b/core/modules/tour/tour.module
@@ -50,7 +50,7 @@ function tour_toolbar() {
     'tab' => [
       '#type' => 'html_tag',
       '#tag' => 'button',
-      '#value' => t('Tour'),
+      '#value' => t('Take a tour of this page'),
       '#attributes' => [
         'class' => ['toolbar-icon', 'toolbar-icon-help'],
         'aria-pressed' => 'false',
@@ -58,8 +58,8 @@ function tour_toolbar() {
       ],
     ],
     '#wrapper_attributes' => [
-      'class' => ['tour-toolbar-tab', 'hidden'],
-      'id' => 'toolbar-tab-tour',
+      'class' => ['tour-toolbar-tab'],
+      'id' => 'toolbar-tab',
     ],
     '#attached' => [
       'library' => [
