diff --git a/core/lib/Drupal/Core/Database/Query/Query.php b/core/lib/Drupal/Core/Database/Query/Query.php
index 37bce1a..651514b 100644
--- a/core/lib/Drupal/Core/Database/Query/Query.php
+++ b/core/lib/Drupal/Core/Database/Query/Query.php
@@ -107,7 +107,8 @@ public function __clone() {
    * Runs the query against the database.
    *
    * @return \Drupal\Core\Database\StatementInterface|null
-   *   A prepared statement, or NULL if the query is not valid.
+   *   A prepared statement, or NULL return for this query,
+   *   if it is not valid.
    */
   abstract protected function execute();
 
